chpl/chpl-auth/src/main/java/gov/healthit/chpl/auth/EmailBuilder.java

```diff
@@ -0,0 +1,167 @@
+package gov.healthit.chpl.auth;
+
+import java.io.File;
+import java.util.Date;
+import java.util.List;
+import java.util.Properties;
+
+import javax.activation.DataHandler;
+import javax.activation.DataSource;
+import javax.activation.FileDataSource;
+import javax.mail.Authenticator;
+import javax.mail.Message.RecipientType;
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMultipart;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.core.env.Environment;
+
+/**
+ * This class is used to send an email.  Properties are set using following a builder pattern.
+ * Sample usage:
+ *      EmailBuilder emailBuilder = new EmailBuilder(env);
+ *      emailBuilder.recipients(recipients)
+ *                  .subject(subject)
+ *                  .htmlMessage(htmlMessage)
+ *                  .fileAttachments(files)
+ *                  .sendEmail();
+ * @author TYoung
+ *
+ */
+public class EmailBuilder {
+    private static final Logger LOGGER = LogManager.getLogger(EmailBuilder.class);
+    private MimeMessage message;
+    private List<String> recipients;
+
+    //optional parameters set to default
+    private String subject = "";
+    private String htmlBody = "";
+    private List<File> fileAttachments = null;
+    private Environment env = null;
+
+    /**
+     * @param env - Spring Environment
+     */
+    public EmailBuilder(final Environment env) {
+        this.env = env;
+    }
+
+
+    /**
+     * Sets the list of recipients for the email.
+     * @param addresses - List of Strings representing email addresses
+     * @return EmailBuilder (this)
+     */
+    public EmailBuilder recipients(final List<String> addresses) {
```

Comment:
These go in the "To:" line of the email, right? I'm wondering if we need a separate builder option for adding recipients to the "Bcc:" line.

Reply:
I did not see any existing functionality to send to BCC or CC.  It can easily be added in the future, but would be difficult to test now, since there there is nowhere in the current app that would use the code.